src/colors.rs

```diff
@@ -0,0 +1,30 @@
+use anyhow::Result;
+use std::io::Write;
+use termcolor::{Color, ColorChoice, ColorSpec, StandardStream, WriteColor};
+
+pub fn print_boldln(txt: &str) -> Result<()> {
+    let mut stdout = StandardStream::stdout(ColorChoice::Auto);
+    stdout.set_color(ColorSpec::new().set_bold(true))?;
+
+    writeln!(&mut stdout, "{}", txt)?;
+    stdout.reset()?;
+    Ok(())
+}
```

Comment:
For a bit more flexibility w.r.t. formatting etc, you could do something like:

```rust
pub fn bold<F>(write: F) -> Result<()>
where
    F: FnOnce(&mut StandardStream),
{
    let mut stdout = StandardStream::stdout(ColorChoice::Auto);
    stdout.set_color(ColorSpec::new().set_bold(true))?;
    write(&mut stdout)?;
    stdout.reset()?;
    Ok(())
}
```
Then you could use it like:

```rust
bold(|s| write!(s, "blah blah: {}", foo))?;
bold(|s| writeln!(s, "writing a line: {}", foo))?;
```

This saves creating the intermediary `String` with `format!` in the case that you need to format some coloured/bolded text.

The colours etc could be handled similarly? We should likely refactor `forc` to use a similar approach.

I think it might be OK to `unwrap` these results following the convention set by `println` (which `unwrap`s on I/O errors internally). Alternatively, the implementation could "fallback" to a normal `println` in the case that any of this custom formatted stream stuff produces an error.

Maybe a better name for this module could be `fmt` as it does more formatting than just colours (i.e. bold).

Reply:
Ah this is super useful - was wondering what the best way to do this would be, thanks for the tips!

At first was trying to see if `tracing` supports colored output but it turns out that the crate doesn't seem to support that and I had to fallback on writes/prints. In situations like [this](https://github.com/FuelLabs/fuelup/pull/64/files#diff-d9079e4f2b5f7f6151c0496aaf01876a092fb845fe2b7a67dd7f9f12d592d905R32-R33) it looks nicer to do `print_with_color()` and then `println!`, even though we already have `info!` from `tracing` - is there a better way to approach this?